py/tests/integration/common.py

```diff
@@ -46,12 +46,20 @@ def external_api(name):
 def internal_api(name):
     return INT_API[name]
 
-def node_online(api):
-    try:
-        top = api.get_top()
-        return top.height > -1
-    except Exception as e:
-        return False
+def node_online(ext_api, int_api):
```

Comment:
is python `and` short circuiting? Not that it is a big deal but if the first check fails the second can be skipped.

Reply:
It is.

From https://docs.python.org/3/reference/expressions.html#boolean-operations
> The expression x and y first evaluates x; if x is false, its value is returned; otherwise, y is evaluated and the resulting value is returned.